src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -75,35 +81,125 @@ internal sealed class HttpWebRequestDiagnosticSource : DiagnosticListener
         private static Func<HttpWebResponse, bool> isWebSocketResponseAccessor;
         private static Func<HttpWebResponse, string> connectionGroupNameAccessor;
 
-        // Fields for controlling initialization of the HttpWebRequestDiagnosticSource singleton
-        private bool initialized = false;
+        internal HttpWebRequestActivitySource()
+        {
+            try
+            {
+                PrepareReflectionObjects();
+                PerformInjection();
+            }
+            catch (Exception ex)
+            {
+                // If anything went wrong, just no-op. Write an event so at least we can find out.
+                InstrumentationEventSource.Log.ExceptionInitializingInstrumentation(typeof(HttpWebRequestActivitySource).FullName, ex);
+            }
+        }
 
-        private HttpWebRequestDiagnosticSource()
-            : base(DiagnosticListenerName)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddRequestTagsAndInstrumentRequest(HttpWebRequest request, Activity activity)
         {
+            activity.DisplayName = BuildDisplayName(request);
```

Comment:
@cijothomas @tarekgh We want to set Activity.DisplayName to what was previously Span.Name right?

Reply:
that is right.